extensions/oidc-client/deployment/src/test/java/io/quarkus/oidc/client/OidcClientTooManyJwtCredentialKeyPropsTestCase.java

```diff
@@ -17,7 +17,8 @@ public class OidcClientTooManyJwtCredentialKeyPropsTestCase {
     static final QuarkusUnitTest test = new QuarkusUnitTest()
             .withApplicationRoot((jar) -> jar
                     .addAsResource(new StringAsset(
-                            "quarkus.oidc-client.token-path=http://localhost:8180/oidc/tokens\n"
+                            "quarkus.keycloak.devservices.enabled=false\n"
+                                    + "quarkus.oidc-client.token-path=http://localhost:8180/oidc/tokens\n"
```

Comment:
Interesting case. This combination is an optimization for any provider which does not support auto-discovery, Instead of typing 3 properties like

```
auth-server-url=some-base-url
discovery-enabled=false
token-path=some-relative-path
```

users just type an absolute URL. I think we need to check for this property as well

Reply:
Please, which number do you mean:

1. we check for `quarkus.oidc-client.token-path` and if it is set, we don't start Keycloak Dev Svc? (I think we started it before when OIDC was present)
2. we check fo `quarkus.oidc-client.token-path` and if it is set, we don't set `quarkus.oidc-client.auth-server-url` to just started Keycloak Dev Svc)?

Or maybe something else @sberyozkin ?

Reply:
Once again I think I misunderstood you (mainly because I don't understand relation to this test you are commenting on), basically you say:

_preset following properties_:

```
auth-server-url=some-base-url
discovery-enabled=false
token-path=some-relative-path
```

but what if discovery should be enabled and auth server URL set? I think current behavior is not breaking. Unless there is something wrong, let's make optimizations later because I'll be frank - I don't understand these properties as you do.

Reply:
Hi @michalvavrik 

>we check for quarkus.oidc-client.token-path and if it is set, we don't start Keycloak Dev Svc? 

yes, unless OIDC is available

> (I think we started it before when OIDC was present)

`quarkus-oidc` does not support omitting `auth-server-url` right now, as it has at least 3 endpoints configured otherwise, this optimization is only for oidc-client (oidc-client-registration)

> we check fo quarkus.oidc-client.token-path and if it is set, we don't set quarkus.oidc-client.auth-server-url to just started Keycloak Dev Svc)?

It should not be started

Reply:
should be done; it seems to address the flakiness, but I am still on fence so I improved error message in these 2 tests to tell us more if this ever happens

Reply:
`discovery-enabled=false` fails too many tests, I am tired of adding `discovery-enabled=true` as I realize you would probably not like it anyway; this is PR that can conflict with others. I'd suggest to make optimizations like this in a separate PR that will be easier to rebase etc. Thank you

Reply:
Of course I'll fix any proposal and fixes you will suggest, I just suggest to leave enhancements and optimizations separately.